### PR TITLE
Add remove_all method

### PR DESCRIPTION
**Pull request by Qew7**
@yegor256 see issue https://github.com/yegor256/zache/issues/6

### CODE REVIEW

File: test/test_zache.rb [hunk 1]
```diff
@@ -83,4 +83,22 @@ def test_remove_key_with_sync_false
     assert(cache.exists?(:hey) == false)
     assert(cache.exists?(:wey) == true)
   end
+
+  def test_remove_all_with_sync
+    cache = Zache.new
```
Comment (yegor256) [inline test/test_zache.rb]: @Qew7 we need a stronger test here. Try to use `Threads` and do `get()` and `remove_all` at the same time in multiple threads. See what happens.

**Comment by Qew7** [comment]
@yegor256 ive remade remove_all logic, can you check it?

**Comment by yegor256** [comment]
@rultor merge

**Comment by yegor256** [comment]
@Qew7 there are merge conflicts, see above. Please resolve them and let me know when the branch is ready. I will try to merge again.

**Comment by Qew7** [comment]
@yegor256 ive resolved conflicts, please try to merge again

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**